src/index/IndexImpl.Text.cpp

```diff
@@ -849,14 +863,43 @@ Index::WordEntityPostings IndexImpl::getContextEntityScoreListsForWords(
                << std::endl;
 
     vector<Index::WordEntityPostings> wepVecs;
+    vector<size_t> skipColumns;
     for (size_t i = 0; i < terms.size(); ++i) {
+      if (!terms[i].ends_with('*')) {
+        skipColumns.push_back(i);
+      }
       if (i != useElFromTerm) {
         wepVecs.push_back(getWordPostingsForTerm(terms[i]));
       }
     }
     wepVecs.push_back(getEntityPostingsForTerm(terms[useElFromTerm]));
     resultWep =
         FTSAlgorithms::crossIntersectKWay(wepVecs, &wepVecs.back().eids_);
+
+    // Restore column order by:
+    // Deleting columns, where the term doesn't end with a '*' and moving the
+    // usElFromTerm column back to its original place
+    vector<vector<WordIndex>> newWidVec;
+    newWidVec.resize(terms.size() - skipColumns.size());
+    size_t j = 0;
+    size_t k = 0;
+    for (size_t i = 0; i < terms.size(); ++i) {
+      if (k < skipColumns.size() && i == skipColumns[k]) {
+        k++;
+      } else {
+        if (i == useElFromTerm) {
+          newWidVec[j] = resultWep.wids_.back();
```

Comment:
The way you do it here is copying the complete vectors.
But if you don't need the old vectors anymore (the ones on the right hand side)
you can move themn like so:
```
newWidVec[j] = std::move(res.wids_.back())
```
(Same for all the other columns below).